common/app/http/PanDomainAuthenticated.scala

```diff
@@ -0,0 +1,138 @@
+package http
+
+import com.amazonaws.services.s3.{AmazonS3, AmazonS3ClientBuilder}
+import com.gu.pandomainauth.PanDomain
+import com.gu.pandomainauth.model.{
+  Authenticated,
+  AuthenticatedUser,
+  Expired,
+  GracePeriod,
+  InvalidCookie,
+  NotAuthenticated,
+  NotAuthorized,
+}
+import common.Environment.{awsRegion, stage}
+import conf.Configuration.aws.mandatoryCredentials
+import scala.concurrent.duration.DurationInt
+import com.gu.pandomainauth.PublicSettings
+import com.gu.permissions.{PermissionDefinition, PermissionsProvider}
+
+trait PanDomainAuthenticated {
```

Comment:
This is explicitly for desktop panda flow, perhaps it could be `PanDomainDesktopAuthentication` or similar?